bread/layout/error.py

```diff
@@ -0,0 +1,40 @@
+from typing import Union
+
+import htmlgenerator as hg
+
+from bread.layout.components.button import Button
+from bread.utils import Link, aslayout
+
+
+@aslayout
+def error_layout(
+    request,
+    status_code: int,
+    status_title: str,
+    description: Union[str, hg.BaseElement],
+    exception_detail: str = None,
+    redirect_link: Link = None,
+):
+    return hg.BaseElement(
+        hg.H1(f"{status_code}: {status_title}", style="margin-bottom: 1rem;"),
+        hg.P(
+            description,
+            style="margin-bottom: 1rem;",
+        ),
+        hg.If(
+            exception_detail,
+            hg.BaseElement(
+                hg.H4("Detail", style="margin-bottom: 1rem;"),
+                hg.DIV(
+                    exception_detail,
+                    style=(
+                        "border: 1px solid grey;"
+                        "padding: 1rem;"
+                        "font-family: monospace;"
+                        "margin-bottom: 1rem;"
+                    ),
+                ),
+            ),
+        ),
+        Button.fromlink(Link(label="Back to homepage", href="/")),
+    )
```

Comment:
One more thing: Could you please move this function into the errors view? Two reasons:
1. It is actually a view and not a layout (the ```@aslayout``` decorator will convert the function into a view)
2. I think you are trying to following the principle of "separation of concerns", right? But you also need to keep in mind the principle of "locality of behaviour" (see https://htmx.org/essays/locality-of-behaviour/ for more about it). In this case, having the layout close to the view makes maintenance easier as the layout will never need to be referenced from other places in the codebase.

Reply:
OK. I'll combine them. Though I didn't know it's called "separation of concerns", but yes, it would make files cleaner for me.